lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -2145,6 +2166,77 @@ def construct_error_details_map_grpc(gax_error)
       {}
     end
 
+    # Construct monitored resource locally for k8s resources.
+    def construct_k8s_resource_locally(local_resource_id)
+      return unless
+        /^
+          (?<resource_type>k8s_container)
+          \.(?<namespace_name>[0-9a-z-]+)
+          \.(?<pod_name>[.0-9a-z-]+)
+          \.(?<container_name>[0-9a-z-]+)$/x =~ local_resource_id ||
+        /^
+          (?<resource_type>k8s_node)
+          \.(?<node_name>[0-9a-z-]+)$/x =~ local_resource_id
+      # Skip for anything that is not k8s related.
+      return unless [K8S_CONTAINER_CONSTANTS[:resource_type],
+                     K8S_NODE_CONSTANTS[:resource_type]].include?(resource_type)
+
+      begin
+        cluster_name = retrieve_k8s_cluster_name
```

Comment:
Might as well reuse your globals (`@k8s_cluster_name` and `@k8s_location`) and inline `retrieve_k8s_cluster_name` and `retrieve_k8s_location`.